### ISSUE TITLE
Ability to generate pre-coded HTML+PHP forms to insert data in database

### ISSUE THREAD

**Issue by akkywadhwa** [phpmyadmin/phpmyadmin#14971]
### Is your feature request related to a problem? Please describe.
The data in these databases is often inserted using any application having a well defined front end and backend. Most developers make an insert form for the tables present in the database. There can be a feature in phpMyAdmin which would generate an application (front end+backend using php) which can prevent developers writing code from the scratch every time.
 
### Describe the solution you'd like
There can be a button which will be activated after selecting a table from the database. After clicking this button, a popup must appear which will ask some parameters required to generate the form. After clicking the submit button on this popup download of the generated form will be started.

### Describe alternatives you've considered
The developers need to make the insertion form from scratch each & everytime.

### Additional context
Deployment guide can also be provided.

- phpMyAdmin version:
All distributions of phpMyAdmin

**Comment by akkywadhwa** [comment]
@williamdes Can you please suggest that how should be the retrieval of column names for generation of forms done? Should it loop over all the column header shown in particular HTML tag or it should fetch information about the table from the database internally?

**Comment by williamdes** [comment]
@akkywadhwa Is this feature request a php code generator ?

**Comment by akkywadhwa** [comment]
@williamdes  Yes! I want to generate boiler-plate code for CRUD operations on the tables of database.

**Comment by williamdes** [comment]
cc @ibennetch Can I have some advice for this feature request please ?

**Comment by OlafvdSpek** [comment]
Would this really make sense for pMA?

**Comment by akkywadhwa** [comment]
@OlafvdSpek pMA is generally used by developers while developing backend and frontend. Imagine a scenario you are developing a website which uses MySQL/Maria DB. You have made a table which stores feedback from all the users of the website. You have made 5 fields in the table to store the feedback. You would need an HTML form with fields of this table and backend code to connect the database to this HTML form. I am proposing a feature which will provide this HTML form and backend code(PHP) as boiler code.
Example:
Sample Table contains 5 fields (First Name, Last Name, Email, Food Quality Rating, Service Rating)

Meaning of boiler-plate code:
HTML forms containing input boxes for all the fields of the table
PHP Code for connecting this HTML form to MySql/Maria DB.

**Comment by ibennetch** [comment]
I think this is a bit beyond the scope of what phpMyAdmin should do. In particular, the HTML part would be customized enough by the programmer that our generic code would probably be meaningless and the PHP part is too complex to make guesses about the user's environment. I think such work is better managed by a good IDE.

In the past, phpMyAdmin had a similar feature, which broke through a code change and we found that the error went unreported by any users for a very long time. In fact, when we looked further in to this, we found that the code that was being generated had been technically correct, but the many different ways programmers were looking to use it meant that the generated code didn't work well for any of their situations, so no one used it anyway.
